src/kvstore/RocksEngine.h

```diff
@@ -54,10 +59,25 @@ class RocksRangeIter : public KVIterator {
     return folly::StringPiece(iter_->value().data(), iter_->value().size());
   }
 
+  rocksdb::Slice* iterateUpperBound() const {
+    return iterateUpperBound_.get();
+  }
+
+  void setIterator(rocksdb::Iterator* iter) {
+    iter_.reset(iter);
+  }
+
+  void setIterator(std::unique_ptr<rocksdb::Iterator> iter) {
+    iter_ = std::move(iter);
+  }
+
  private:
-  std::unique_ptr<rocksdb::Iterator> iter_;
+  std::unique_ptr<rocksdb::Iterator> iter_{nullptr};
   rocksdb::Slice start_;
   rocksdb::Slice end_;
+  // Make sure that the lifetime of iterate_upper_bound in rocksdb::ReadOptions
+  // is the same as RocksRangeIter object
+  std::unique_ptr<rocksdb::Slice> iterateUpperBound_;
```

Comment:
just `uppperBound_` is ok, same reason as above. Besides, could keep consist type with start and end, that it, just use `rocksdb::Slice`.